packages/backend/src/plugins/auth.ts

```diff
@@ -23,5 +27,28 @@ export default async function createPlugin({
   config,
   discovery,
 }: PluginEnvironment) {
-  return await createRouter({ logger, config, database, discovery });
+  return await createRouter({
+    logger,
+    config,
+    database,
+    discovery,
+    providerFactories: {
+      ...defaultAuthProviderFactories,
+      google: createGoogleProvider({
+        signIn: {
+          // resolver: 'email',
+          resolver: async ({ profile: { email } }, ctx) => {
+            if (!email) {
+              throw new Error('No email associated with user account');
+            }
+            const id = email.split('@')[0];
+            const token = await ctx.tokenIssuer.issueToken({
+              claims: { sub: id, ent: [id] },
+            });
+            return { id, token };
```

Comment:
would it make sense to return { id, claims } instead and let the framework issue the token, or are there use cases where more than just the claims are to be "tweaked" by the resolver?

Reply:
answered: no, let's keep this as is